pkg/client/typed/dynamic/client.go

```diff
@@ -251,6 +245,18 @@ func (dynamicCodec) Encode(obj runtime.Object, w io.Writer) error {
 	return runtime.UnstructuredJSONScheme.Encode(obj, w)
 }
 
+// ContentConfig returns a restclient.ContentConfig for dynamic types.
+func ContentConfig() restclient.ContentConfig {
```

Comment:
hard-coding api.Codecs into this method is fairly wrong - especially since you use nothing but JSON.  You should just be using your own CodecFactory here and only register json.